pkg/secrethub/client.go

```diff
@@ -110,7 +115,19 @@ func NewClient(with ...ClientOption) (*Client, error) {
 
 	// Try to use default key credentials if none provided explicitly
 	if client.decrypter == nil {
-		err := client.with(WithCredentials(credentials.UseKey(client.DefaultCredential())))
+		identityProvider := os.Getenv("SECRETHUB_IDENTITY_PROVIDER")
+
+		var provider credentials.Provider
+		switch identityProvider {
+		case "key", "":
```

Comment:
Should this be case insensitive?

Reply:
Also after looking at the code for a while I now see you have the `""` empty string match here too. That's not clear at a glance. Looking at this switch it looks like the default is to error. Can we make this a bit more readable?

Reply:
> Should this be case insensitive?

Yes, good suggestion. I'll also submit a PR to [secrethub/secrethub-cli](https://github.com/secrethub/secrethub-cli) to make the envvar and flag case insensitive there as well.

Reply:
secrethub/secrethub-cli#160

Reply:
> 

Also after looking at the code for a while I now see you have the "" empty string match here too. That's not clear at a glance. Looking at this switch it looks like the default is to error. Can we make this a bit more readable?

I've swapped the `""` and `"key"` values to make the empty string stand out more. Does this help?